src/inmanta/protocol/openapi/model.py

```diff
@@ -65,11 +67,30 @@ class Schema(BaseModel):
     readOnly: Optional[bool] = None
     example: Optional[Any] = None
     deprecated: Optional[bool] = None
-    anyOf: Optional[Sequence["Schema"]] = None
-    allOf: Optional[Sequence["Schema"]] = None
-    oneOf: Optional[Sequence["Schema"]] = None
+    anyOf: Optional[abc.Sequence["Schema"]] = None
+    allOf: Optional[abc.Sequence["Schema"]] = None
+    oneOf: Optional[abc.Sequence["Schema"]] = None
     enum: Optional[List[str]] = None
 
+    @pydantic.model_validator(mode="after")
+    def convert_null_any_of(self: Self) -> Self:
+        """
+        Convert null in anyOf to the `nullable` property.
+        """
+        if self.anyOf is not None:
+            without_null: abc.Sequence["Schema"] = [e for e in self.anyOf if e.type != "null"]
+            if len(without_null) != len(self.anyOf):
+                if len(without_null) == 1:
+                    # promote single child, which has already been validated at this point
+                    child: "Schema" = without_null[0]
+                    for field in child.model_fields_set:
+                        setattr(self, field, getattr(child, field))
```

Comment:
I'm not a huge fan of this but it's the best/cleanest approach I could think of. Suggestions welcome.

Reply:
I don't fully understand what we're trying to achieve here, I'm not familiar with openapi

Reply:
The same for me. This needs more documentation.